### DISCUSSION TITLE
Context Menus

### DISCUSSION THREAD

**Discussion by goaaats** [goatcorp/Dalamud#746] [Announcements]
Dalamud can now do Context Menu modifications for you, with special support for players, basically everywhere you can right click them, and items. You can add custom context menu options everywhere.

There's a problem: you can't use this together with the other context menu modification methods, such as xivcommon. That's why it's only on testing right now. We need to coordinate to bring this to a release without increasing the API level.
There is a new testing key for this purpose, it is: `proof of context`

The `ContextMenu` class is registered as a plugin service. [See here for usage.](https://github.com/goatcorp/Dalamud/blob/master/Dalamud/Interface/Internal/Windows/SelfTest/AgingSteps/ContextMenuAgingStep.cs#L150-L232)

All Dalamud context menu items are forced to have a red "D" at the front. This is because people are now reporting bugs for plugins on OF because they think the option is from the game.

![image](https://user-images.githubusercontent.com/16760685/151647182-fc202f1d-9323-4750-bc39-3504c79edbb0.png)

**Comment by goaaats** [comment]
Plugins that need to adjust:
* NoSoliciting
* MarketBoardPlugin
* SimpleTweaks
* PlayerTrack
* Visibility (PRd)
* Party Icons
* Player Tags
* PriceCheck

Please PR updates to those plugins, and we'll merge them all at once.

**Reply by kalilistic** [reply]
PriceCheck too.
